guiEditor/src/diagram/workbench.tsx

```diff
@@ -90,6 +96,32 @@ export class WorkbenchComponent extends React.Component<IWorkbenchComponentProps
         return this._selectedGuiNodes;
     }
 
+    // given a control gets the parent up the tree selectionDepth times. Selection depth is altered as we go down the tree.
+    private _getParentWithDepth(control: Control) {
+        --this._selectionDepth;

```

Comment:
should this be part of the loop ? as it looks like here it will always only remove 1 independently of the depth ?

Reply:
No. It should only remove once per functional call.